test/SerialIOTest.cpp

```diff
@@ -4203,6 +4213,176 @@ TEST_CASE( "variableBasedParticleData", "[serial][adios2]" )
 }
 #endif
 
+#if openPMD_HAVE_ADIOS2
+#ifdef ADIOS2_HAVE_BZIP2
+TEST_CASE( "automatically_deactivate_span", "[serial][adios2]" )
+{
+    // automatically (de)activate span-based storeChunking
+    {
+        Series write( "../samples/span_based.bp", Access::CREATE );
+        auto E_uncompressed = write.iterations[ 0 ].meshes[ "E" ][ "x" ];
+        auto E_compressed = write.iterations[ 0 ].meshes[ "E" ][ "y" ];
+
+        Dataset ds{ Datatype::INT, { 10 } };
+
+        E_uncompressed.resetDataset( ds );
+
+        std::string compression = R"END(
+{
+  "adios2": {
+    "dataset": {
+      "operators": [
+        {
+          "type": "bzip2"
+        }
+      ]
+    }
+  }
+})END";
+
+        ds.options = compression;
+        E_compressed.resetDataset( ds );
+
+        bool spanWorkaround = false;
+        E_uncompressed.storeChunk< int >(
+            { 0 }, { 10 }, [ &spanWorkaround ]( size_t size ) {
+                spanWorkaround = true;
+                return std::shared_ptr< int >(
+                    new int[ size ]{}, []( auto * ptr ) { delete[] ptr; } );
+            } );
+
+        REQUIRE( !spanWorkaround );
+
+        E_compressed.storeChunk< int >(
+            { 0 }, { 10 }, [ &spanWorkaround ]( size_t size ) {
+                spanWorkaround = true;
+                return std::shared_ptr< int >(
+                    new int[ size ]{}, []( auto * ptr ) { delete[] ptr; } );
+            } );
+
+        REQUIRE( spanWorkaround );
```

Comment:
@franzpoeschel does this test require the new JSON backend or so? In that case, I would need to remove it from the backport.

Reply:
It does, though it might actually make sense to backport the workaround too. I'll have a look

Reply:
I've opened the backport as a PR: https://github.com/ax3l/openPMD-api/pull/2